internal/pkg/service/buffer/storage/local/writer/csv/csv_test.go

```diff
@@ -0,0 +1,165 @@
+package csv_test
+
+import (
+	"compress/gzip"
+	"fmt"
+	"github.com/c2h5oh/datasize"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/storage"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/storage/compression"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/storage/local/writer/disksync"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/storage/local/writer/test/testcase"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/store/model/column"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
+	"github.com/klauspost/compress/zstd"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"io"
+	"strings"
+	"testing"
+	"time"
+)
+
+type fileCompression struct {
+	Name        string
+	Config      compression.Config
+	FileDecoder func(t *testing.T, r io.Reader) io.Reader
+}
+
+func TestCSVWriter(t *testing.T) {
+	t.Parallel()
```

Comment:
`TestCSVWriter` testuje vsetky zakladne kombinacie nastaveni, pomohlo mi to najst este nejake chyby:

![image](https://github.com/keboola/keboola-as-code/assets/19371734/f59272ff-0d81-4e88-86bd-ccb45fdba0e9)